src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/UnsynchronizableContentDocumentProcessedListener.cs

```diff
@@ -73,14 +72,10 @@ public override void DocumentProcessed(DocumentSnapshot document)
             {
                 // Documents are identical but we didn't synchronize them because they didn't need to be re-evaluated.
 
-                var request = new UpdateCSharpBufferRequest()
-                {
-                    HostDocumentFilePath = document.FilePath,
-                    Changes = Array.Empty<TextChange>(),
-                    HostDocumentVersion = syncVersion
-                };
+                var result = document.TryGetText(out var latestText);
+                Debug.Assert(result, "We just successfully retrieved the text version, this should always return true.");
```

Comment:
Love including little sanity checks like this. Is there any more context on what returning false from `TryGetText` might mean? I tried looking at [the source](https://github.com/aspnet/AspNetCore-Tooling/blob/a8992d05a0b52e737a632bdae407395397898371/src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/DocumentState.cs#L133) but it's a bit obtuse to someone who's still learning the area.

Reply:
Only things I could think of is that our underlying Razor representation changed the Roslyn SourcrText unexpectedly